flink-table/flink-table-common/src/test/java/org/apache/flink/table/types/TypeInfoDataTypeConverterTest.java

```diff
@@ -0,0 +1,262 @@
+
+    @Parameters(name = "{index}: {0}")
+    public static List<TestSpec> testData() {
+        return Arrays.asList(
+                TestSpec.forType(Types.INT).expectDataType(DataTypes.INT().notNull()),
+                TestSpec.forType(Types.BIG_DEC)
+                        .expectDataType(DataTypes.DECIMAL(38, 18).nullable()),
+                TestSpec.forType(
+                                Types.ROW_NAMED(
+                                        new String[] {"a", "b", "c"},
+                                        Types.DOUBLE,
+                                        Types.INT,
+                                        Types.STRING))
+                        .expectDataType(
+                                DataTypes.ROW(
+                                                DataTypes.FIELD("a", DataTypes.DOUBLE()),
+                                                DataTypes.FIELD("b", DataTypes.INT()),
+                                                DataTypes.FIELD("c", DataTypes.STRING()))
+                                        .notNull()),
+                TestSpec.forType(
+                                Types.TUPLE(
+                                        Types.INT,
+                                        Types.TUPLE(Types.STRING, Types.BOOLEAN),
+                                        Types.LONG))
+                        .expectDataType(
+                                DataTypes.STRUCTURED(
+                                                Tuple3.class,
+                                                DataTypes.FIELD("f0", DataTypes.INT().notNull()),
+                                                DataTypes.FIELD(
+                                                        "f1",
+                                                        DataTypes.STRUCTURED(
+                                                                        Tuple2.class,
+                                                                        DataTypes.FIELD(
+                                                                                "f0",
+                                                                                DataTypes.STRING()
+                                                                                        .nullable()),
+                                                                        DataTypes.FIELD(
+                                                                                "f1",
+                                                                                DataTypes.BOOLEAN()
+                                                                                        .notNull()))
+                                                                .notNull()),
+                                                DataTypes.FIELD("f2", DataTypes.BIGINT().notNull()))
+                                        .notNull()),
+                TestSpec.forType(Types.POJO(PojoWithFieldOrder.class))
```

Comment:
I think something is wrong with POJO types mapping. Shouldn't the primitive fields be mapped to `NOT NULL` types?
Also from our offline chat it seems bridging classes are also not mapped correctly.